lib/ansible/plugins/lookup/file.py

```diff
@@ -25,6 +25,14 @@
         type: bool
         required: False
         default: False
+      encodings:
+        description: encoding types of files to be read
+        default: ['utf-8', 'utf-16']
+        type: list
+      warnings:
```

Comment:
Lookups already support a generic way to handle errors, we don't need additional overlapping functionality.  See `errors` under https://docs.ansible.com/ansible/latest/plugins/lookup.html#using-lookup-plugins

Reply:
@sivel In order to support functionality as I wanted I don't raise UnicodeDecodeErrors as they occur. In order to still make this available to the end user I was exposing an optional parameter so that they get the decode errors as they occur.

Otherwise you simply get, 
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'file'. Error was a <class 'ansible.errors.AnsibleError'>, original message: Failed to decode file using [utf8, utf32] encoding types."}

Which I feel is less informative, I can strip it if you feel deeply that it's unneeded tough.

Reply:
TASK [debug] **************************************************************************************************************************************************************************************
 [WARNING]: 'utf8' codec can't decode byte 0xff in position 0: invalid start byte

 [WARNING]: 'utf32' codec can't decode bytes in position 0-3: codepoint not in range(0x110000)

from ansible.errors import AnsibleError, AnsibleParserError
ok: [localhost] => {
    "msg": "This is a test."
}

PLAY RECAP ****************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

Reply:
You can always catch the exceptions and do that.

```
try:
    contents = to_text(b_contents, encoding=encoding, errors='surrogate_or_strict')
except UnicodeDecodeError:
    raise AnsibleError('Failed to decode file using %s encoding' % encoding)
except Exception:
    raise
```

Encoding needs to be explicitly singular, and no duplication of functionalities.

Reply:
@sivel, I see what your saying now. I will tie it into errors='warn' instead of an additional option.

Reply:
@sivel, I removed the warnings option and recommitted. I don't believe that errors='warn' gives the same functionality but I can understand you point on the potential for confusion. I think people should get the idea from the returned error.